File: codeforces/_1490e.java
import java.util.*;
import java.io.*;

public class _1490e {
    public static void main(String[] args) throws IOException {
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(in.readLine());
        
        for (int i = 0; i < t; i++) {
            int n = Integer.parseInt(in.readLine());
            String[] nums = in.readLine().split(" ");
            int[][] a = new int[n][2];
            for (int j = 0; j < n; j++) {
                a[j][0] = Integer.parseInt(nums[j]);
                a[j][1] = j+1;
            }
            Arrays.sort(a, (x, y) -> {
                return y[0] - x[0];
            });
            
            int left = a[0][0];
            int prev = 0;
            int pending = 0;
            long total = 0;
            for (int[] num : a) total += num[0];
            ArrayList<Integer> ans = new ArrayList<Integer>();
            for (int j = 0; j < n; j++) {
                if (a[j][0] != prev) {
                    total -= pending;
                    pending = a[j][0];
                } else {
                    pending += prev;
                }
                prev = a[j][0];

                if (total >= left) {
                    ans.add(a[j][1]);
                    left = a[j][0];
                } else {
                    break;
                }
            }

            Collections.sort(ans);
            System.out.println(ans.size());
            for (int id : ans) {
                System.out.print(id + " ");
            }
            System.out.println();
        }
    }
}

File: codeforces/1490e.py
from sys import stdin, stdout, setrecursionlimit

setrecursionlimit(200000)

def _i():
    return stdin.readline().strip()

def _p(x, end='\n'):
    stdout.write(str(x))
    stdout.write(end)

for t in range(int(_i())):
    n = int(_i())
    a = map(int, _i().split())
    a = sorted([(num, i+1) for i, num in enumerate(a)], reverse=True)

    total = sum([a[i][0] for i in range(n)])
    prev = a[0][0]
    pending = 0
    totals = [0]*n
    for i in range(n):
        if a[i][0] != prev:
            total -= pending
            pending = 0
        pending += a[i][0]
        totals[i] = total


    

stdout.flush()